firebase-firestore/src/main/java/com/google/firebase/firestore/remote/AbstractStream.java

```diff
@@ -174,6 +174,9 @@ public void run() {
   /** The time a stream stays open after it is marked idle. */
   private static final long IDLE_TIMEOUT_MS = TimeUnit.MINUTES.toMillis(1);
 
+  /** Maximum backoff time when reconnecting. */
```

Comment:
"when reconnecting" doesn't really do anything to distinguish this from the existing BACKOFF_MAX_DELAY_MS which is also used when reconnecting.

I think the intention is that this max backoff delay is used when we're 100% sure that the connection failed on the client-side and didn't even reach the server.  In practice, I think the only case we can be sure of that is DNS failures (see my other comment about ConnectException).  If that's the case, then maybe we can call this DNS_FAILURE_BACKOFF_MAX_DELAY_MS or something...

Reply:
How about `CLIENT_NETWORK_FAILURE_BACKOFF_MAX_DELAY_MS`?

Reply:
I just noticed that BACKOFF should maybe be first (to group this with the other backoff-related constants).  So: `BACKOFF_CLIENT_NETWORK_FAILURE_MAX_DELAY_MS`?

Reply:
FWIW- I don't feel strongly.  Just trying to figure out how to make the name parse more sensibly...

Reply:
Changed to Michael's suggestion.